File: ruby/eyehub_get_events_example.rb
#!/usr/bin/ruby
# Simple example script to retrieve the events for the device
# named in DEVICE_ID using the EyeHub API

require 'rubygems'
require 'json'
require 'net/https'

# Change these to match your account and device details
EYEHUB_USERNAME="YOUR_EYEHUB_USERNAME"
EYEHUB_PASSWORD="YOUR_EYEHUB_PASSWORD"
DEVICE_MANAGER_ID="DEVICE_MANAGER_ID"
DEVICE_ID="YOUR_DEVICE_ID"

# Now let's get a list of just the events
puts "Retrieving a list of events..."
event_uri = URI.parse("https://config28.flexeye.com/v1/iot_Default/dms/#{DEVICE_MANAGER_ID}/devices/#{DEVICE_ID}/events")
http = Net::HTTP.new(event_uri.host, event_uri.port)
http.use_ssl = true
request = Net::HTTP::Get.new(event_uri.request_uri)
request.basic_auth(EYEHUB_USERNAME, EYEHUB_PASSWORD)
event_data = http.request(request)
# Again, the response will be formatted in JSON
event_details = JSON.parse(event_data.body)
puts "Number of events found: "+event_details["events"].count.to_s
# And output some info for each event we found
event_details["events"].each do |ev|
  puts ev["uri"]
end
puts

File: python/eyehub_get_events_example.py
#!/usr/bin/python
# Simple example script to retrieve the events for the device
# named in DEVICE_ID using the EyeHub API

import requests
from requests.auth import HTTPBasicAuth
import json

# Change these to match your account and device details
EYEHUB_USERNAME="YOUR_EYEHUB_USERNAME"
EYEHUB_PASSWORD="YOUR_EYEHUB_PASSWORD"
DEVICE_MANAGER_ID="DEVICE_MANAGER_ID_FOR_DEVICE_IN_DEVICE_ID"
DEVICE_ID="YOUR_DEVICE_ID"

print "Retrieving a list of events..."

url = "https://config28.flexeye.com/v1/iot_Default/dms/%s/devices/%s/events" % (DEVICE_MANAGER_ID, DEVICE_ID)

# Post to the URL after encoding the dictionary as json
response = requests.get(url, auth=(EYEHUB_USERNAME, EYEHUB_PASSWORD))

if response.status_code == 200:
    # Load response content into a dict from json
    response_data = json.loads(response.content)
    print "Number of events found: %i" % len(response_data['events'])
    for ev in response_data['events']:
        print ev['uri']
else:
    print response.status_code
    print response

